src/Corax/Indexing/IndexWriter.cs

```diff
@@ -1275,7 +1306,17 @@ public void InsertTextualField()
                 }
                 
                 RecordTermsForEntries(_entriesForTerm, entries, termContainerId);
-
+    
+                //Update mapping virtual<=> storage location location. Final writing will be done after inserting ALL terms for specific field.
+                if (_indexedField.FieldIndexingMode is FieldIndexingMode.Search)
+                {
+                    ref var virtualMapping = ref _virtualTermIdToTermContainerId;
+                    Debug.Assert(virtualMapping.Count == _indexedField.Textual.Count(), "virtualMapping.Count == _indexedField.Textual.Count()");
+                    Debug.Assert(virtualMapping[entries.StorageLocation] != Constants.IndexedField.Invalid, "virtualMapping[entries.StorageLocation] != Constants.IndexedField.Invalid, Term doesn't exists on disk!");
+                    virtualMapping[entries.StorageLocation] = termContainerId;
```

Comment:
Why do we need the `StorageLocation`, when we get the `entries`, we already know what the index is, no?
That can save us some allocated space / need to track things